### PR TITLE
Feature/acs 1787 endpoint for deleted nodes

### PR DESCRIPTION
**Pull request by mikolajbrzezinski**
https://alfresco.atlassian.net/browse/ACS-1787

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java [hunk 1]
```diff
@@ -34,6 +34,8 @@
 import org.alfresco.rest.framework.resource.content.BinaryResource;
 import org.alfresco.rest.framework.resource.parameters.CollectionWithPagingInfo;
 import org.alfresco.rest.framework.resource.parameters.Parameters;
+import org.alfresco.service.cmr.repository.DirectAccessUrl;
+import org.alfresco.service.cmr.repository.NodeRef;
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java:38]: you can remove this line as it is an unused import

File: remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java [hunk 2]
```diff
@@ -99,4 +101,29 @@
      * @return
      */
     CollectionWithPagingInfo<Rendition> getRenditions(String archivedId, Parameters parameters);
+
+    /**
+    * @author Mikołaj Brzeziński
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java:106]: no need to add `@author` for individual methods

File: remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java [hunk 3]
```diff
@@ -99,4 +101,29 @@
...
+    * Gets a presigned URL to directly access content.
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java:107]: maybe add a blank line between the description and the parameters

File: remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java [hunk 4]
```diff
@@ -99,4 +101,29 @@
...
+    * @param archivedId The node id for which to obtain the direct access {@code URL}
+    * @param attachment {@code true} if an attachment {@code URL} is requested, {@code false} for an embedded {@code URL}, {@code true} by default.
+    * @return A direct access {@code URL} object for the content.
+    */
+    default DirectAccessUrl requestContentDirectUrl(String archivedId, boolean attachment)
+    {
+        return requestContentDirectUrl(archivedId, attachment,null);
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java:114]: add a space between the comma and `null`

File: remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java [hunk 5]
```diff
@@ -99,4 +101,29 @@
...
+     * Gets a presigned URL to directly access content.
+     * @param archivedId The node id for which to obtain the direct access {@code URL}
+     * @param attachment {@code true} if an attachment {@code URL} is requested, {@code false} for an embedded {@code URL}, {@code true} by default.
+     * @param validFor The time at which the direct access {@code URL} will expire.
+     * @return A direct access {@code URL} object for the content.
+     */
+    DirectAccessUrl requestContentDirectUrl(String archivedId, boolean attachment, Long validFor);
+
+
+
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/DeletedNodes.java:128]: maybe remove at least 2 of these blank lines

File: remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java [hunk 6]
```diff
@@ -48,13 +48,15 @@
 import org.alfresco.rest.api.model.UserInfo;
 import org.alfresco.rest.framework.core.exceptions.ApiException;
 import org.alfresco.rest.framework.core.exceptions.ConstraintViolatedException;
+import org.alfresco.rest.framework.core.exceptions.DisabledServiceException;
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java:51]: unused import

File: remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java [hunk 7]
```diff
@@ -244,4 +246,21 @@ public Rendition getRendition(String archivedId, String renditionId, Parameters
         NodeRef nodeRef = new NodeRef(StoreRef.STORE_REF_ARCHIVE_SPACESSTORE, archivedId);
         return renditions.getRenditions(nodeRef, parameters);
     }
+
+
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java:250]: remove extra blank line

File: remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java [hunk 8]
```diff
@@ -244,4 +246,21 @@ public Rendition getRendition(String archivedId, String renditionId, Parameters
...
+    /**
+     * @author Mikołaj Brzeziński
+     * Gets a presigned URL to directly access content.
+     * @param originalNodeId The node id for which to obtain the direct access {@code URL}
+     * @param attachment {@code true} if an attachment {@code URL} is requested, {@code false} for an embedded {@code URL}, {@code true} by default.
+     * @param validFor The time at which the direct access {@code URL} will expire.
+     * @return A direct access {@code URL} object for the content.
+     */
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java:258]: you can replace these javadoc comments with:
```
    /**
     * {@inheritDoc}
     */
```
which will inherit the javadoc comments from the interface, removing the need to update in both places if/when the comments change :)

File: remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java [hunk 9]
```diff
@@ -244,4 +246,21 @@ public Rendition getRendition(String archivedId, String renditionId, Parameters
...
+    @Override
+    public DirectAccessUrl requestContentDirectUrl(String originalNodeId, boolean attachment, Long validFor)
+    {
+        //First check the node is valid and has been archived.
+        NodeRef validatedNodeRef = nodes.validateNode(StoreRef.STORE_REF_ARCHIVE_SPACESSTORE, originalNodeId);
+        return nodes.requestContentDirectUrl(validatedNodeRef,attachment,validFor);
```
Comment (SaraAspery) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/DeletedNodesImpl.java:264]: maybe just add a space between each parameter

**Review by SaraAspery — CHANGES_REQUESTED** [review]
This looks really good Mikolaj! There are just a few very minor style/comment changes to make. Well done :)

**Review by SaraAspery — APPROVED** [review]

**Review by alandavis — COMMENTED** [review]
LGTM - nice and clean

**Pull request merged**